### DISCUSSION TITLE
OpenTelemetry and Activemq

### DISCUSSION THREAD

**Discussion by mmorel-35** [open-telemetry/opentelemetry-java-instrumentation#9680] [Q&A]
In the actual document I couldn't find anything concerning context propagation when using Activemq.
In HTTP context, I'm using Zipkin headers for context propagation but when it comes to Activemq I am clueless. As the Actual documentation on OpenTelemetry  is linking two documents but without saying more 
And on Activemq sidew there is only the metrics plugin that is available.
Moreover having the OpenTelemetry java agent  deployed along the Activemq doesn't seems to provide anything as it does for tomcat for example.

Anyone with experience could help on this ?
Shall the parentId and the spanId be provided as a JMS header ?
With which case ?

**Comment by trask** [comment]
> Moreover having the OpenTelemetry java agent deployed along the Activemq doesn't seems to provide anything as it does for tomcat for example.

hi @mmorel-35, the ActiveMQ instrumentation is for the ActiveMQ Java client libraries (not the ActiveMQ server itself), could this be the source of issue?

**Reply by mmorel-35** [reply]
Hi @trask, 
I understood it by experience indeed. 
My main goel behind this question how do you handle OpenTelemetry Tracing feature with an OpenTelemetry agent on the consumer and one on the Producer of the events with JMS Client library ?

**Reply by laurit** [reply]
we pass the trace context via jms headers

**Reply by adben** [reply]
Can we please have some snippets for your suggestion @laurit ?

**Reply by laurit** [reply]
context propagation header is set in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ab09fcee98c3b86993c1de303d09055a91b623eb/instrumentation/jms/jms-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/MessagePropertySetter.java#L21 and read in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ab09fcee98c3b86993c1de303d09055a91b623eb/instrumentation/jms/jms-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/MessagePropertyGetter.java#L24
